python/test/pyJoint_TEST.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (C) 2022 Open Source Robotics Foundation
+
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from ignition.math import Pose3d, Vector3d
+from sdformat import Joint, JointAxis, Error, SemanticPose
+import math
+import unittest
+
+class JointTEST(unittest.TestCase):
+
+    def test_default_construction(self):
+        joint = Joint()
+        self.assertFalse(joint.name())
+        self.assertEqual(Joint.JointType.INVALID, joint.type())
+        self.assertFalse(joint.parent_link_name())
+        self.assertFalse(joint.child_link_name())
+        self.assertEqual(Pose3d.ZERO, joint.raw_pose())
+        self.assertFalse(joint.pose_relative_to())
+
+        semanticPose = joint.semantic_pose()
+        self.assertEqual(Pose3d.ZERO, semanticPose.raw_pose())
+        self.assertFalse(semanticPose.relative_to())
+        pose = Pose3d()
+        # expect errors when trying to resolve pose without graph
+        self.assertTrue(semanticPose.resolve(pose))
+
+        joint.set_raw_pose(Pose3d(-1, -2, -3, math.pi, math.pi, 0))
+        self.assertEqual(Pose3d(-1, -2, -3, math.pi, math.pi, 0),
+                joint.raw_pose())
+
+        joint.set_pose_relative_to("link")
+        self.assertEqual("link", joint.pose_relative_to())
+
+        semanticPose = joint.semantic_pose()
+        self.assertEqual(joint.raw_pose(), semanticPose.raw_pose())
+        self.assertEqual("link", semanticPose.relative_to())
+        pose = Pose3d()
+        # expect errors when trying to resolve pose without graph
+        self.assertTrue(semanticPose.resolve(pose))
+
+        joint.set_name("test_joint")
+        self.assertEqual("test_joint", joint.name())
+
+        joint.set_parent_link_name("parent")
+        self.assertEqual("parent", joint.parent_link_name())
+
+        joint.set_child_link_name("child")
+        self.assertEqual("child", joint.child_link_name())
+
+        body = ""
+        resolveChildLink = joint.resolve_child_link(body)
+        self.assertEqual(1, len(resolveChildLink[0]))
+        self.assertFalse(resolveChildLink[1])
+        resolveParentLink = joint.resolve_parent_link(body)
+        self.assertEqual(1, len(resolveParentLink[0]))
+        self.assertFalse(resolveParentLink[1])
```

Comment:
These should use the new `resolve_(child|parent)_link` API

Reply:
I think I'm using the new tuple, what else is required?